src/App.tsx

```diff
@@ -21,75 +23,104 @@ enum SortType {
-export function getReorderedGoods(
-  goods: string[],
-  { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+export class App extends Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  reverseGoods = () => {
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }));
+  };
 
-  return visibleGoods;
-}
+  sortGoodsAlphabetic = () => {
+    this.setState(() => ({
+      sortType: SortType.ALPHABET,
+    }));
+  };
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
+  resetState = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
 
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
+  sortGoodsByLength = () => {
+    this.setState(() => ({
+      sortType: SortType.LENGTH,
+    }));
+  };
 
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
+  render() {
+    const { isReversed, sortType } = this.state;
 
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
+    getReorderedGoods(goodsFromServer, this.state);
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames(
+              'button is-info', {
+                'is-light': sortType !== SortType.ALPHABET,
+              },
+            )}
+            onClick={this.sortGoodsAlphabetic}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button is-success', {
+                'is-light': sortType !== SortType.LENGTH,
+              },
+            )}
+            onClick={this.sortGoodsByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button is-warning', {
+                'is-light': !isReversed,
+              },
+            )}
+            onClick={this.reverseGoods}
+          >
+            Reverse
+          </button>
+          {(isReversed || sortType !== SortType.NONE) && (
```

Comment:
Let's move this to a variable, to make it more readable

Reply:
done